### PR TITLE
Renaming subplaylists

### PR DESCRIPTION
**Pull request by CarlinWilliamson**
## Context
[Corresponding Pivotal Tracker story](https://www.pivotaltracker.com/story/show/175033811)

## Description 💬
Added a modal for changing a sub-playlist's name

Note: there is nothing stopping two sub-playlists having the same name. I think this should be dealt with when saving the playlists

## Checklist ✅:

- [x] I assigned the story to myself on Pivotal Tracker before I started working on this
- [x] Added a link to the user story context
- [x] Added a description of my changes
- [x] Assigned at least two reviewers

### CODE REVIEW

File: src/components/EditPlaylistNameDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+
+export default function FormDialog(props: { name: string, closeAndSet: (newName: string) => void }) {
```
Comment (featherbear) [inline src/components/EditPlaylistNameDialog.tsx]:
```suggestion
export default function FormDialog(props: { name: string, onClose: (newName: string) => void }) { 
```

File: src/components/EditPlaylistNameDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        <TextField
+          autoFocus
+          label="New Name"
+          defaultValue={props.name}
+          fullWidth
+          onChange={(e) => setNewName(e.target.value)}
+        />
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={() => props.closeAndSet(props.name)} color="primary">
```
Comment (featherbear) [inline src/components/EditPlaylistNameDialog.tsx]:
```suggestion
        <Button onClick={() => props.onClose(props.name)} color="primary">
```

File: src/components/EditPlaylistNameDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          Cancel
+      </Button>
+        <Button onClick={() => props.closeAndSet(newName)} color="primary">
```
Comment (featherbear) [inline src/components/EditPlaylistNameDialog.tsx]:
```suggestion
        <Button onClick={() => props.onClose(newName)} color="primary">
```

File: src/components/Subplaylist.tsx [hunk 4]
```diff
@@ -27,44 +33,62 @@ export default function Subplaylist(props: {
...
-          {props.playlist.tracks.map((track: TrackObj) => (
-            <TableRow key={track.id}> 
-            {/* UUID for each track item */}
-              <TableCell component="th" scope="row">
-                <Track track={track} />
+    <div>
+      <Dialog open={editDialogOpen} onClose={() => setEditDialogOpen(false)}>
+        <EditPlaylistNameDialog
+          name={props.playlist.name}
+          closeAndSet={(newName: string) => {
```
Comment (featherbear) [inline src/components/Subplaylist.tsx]:
```suggestion
          onClose={(newName: string) => {
```

**Review by featherbear — CHANGES_REQUESTED** [review]
Could you rename your event handler prop to something more canonical like `onClose`

**Comment by CarlinWilliamson** [comment]
> Could you rename your event handler prop to something more canonical like `onClose`

We can (would do it in a single commit).
I was purposely stearing away from a normal name because it does extra things...
I don't do much frontend. So perhaps the more canonical name is better

**Comment by featherbear** [comment]
> 
> 
> > Could you rename your event handler prop to something more canonical like `onClose`
> 
> We can (would do it in a single commit).
> I was purposely stearing away from a normal name because it does extra things...

Perhaps `onSave` then.
And you could even fire the callback as `onSave(null)` if the Cancel button was pressed instead?

Otherwise React < may > fire an unnecessary component render

**Review by featherbear — APPROVED** [review]
lgtm

**Review by z5160345 — APPROVED** [review]

**Pull request merged**